src/shared/test/frame-rpc-test.js

```diff
@@ -89,58 +91,77 @@ describe('shared/bridge', () => {
     });
   });
 
-  it('should ignore invalid messages', done => {
-    // Correct message
+  it('should call method on valid message', done => {
     port1.postMessage({
       arguments: [1, 2],
       method: 'plusOne',
       protocol: 'frame-rpc',
       version: '1.0.0',
     });
 
-    // Incorrect argument
-    port1.postMessage({
-      arguments: 'test',
-      method: 'plusOne',
-      protocol: 'frame-rpc',
-      version: '1.0.0',
-    });
-
-    // Incorrect method
-    port1.postMessage({
-      arguments: [1, 2],
-      method: 'dummy',
-      protocol: 'frame-rpc',
-      version: '1.0.0',
-    });
-
-    // Incorrect protocol
-    port1.postMessage({
-      arguments: [1, 2],
-      method: 'plusOne',
-      protocol: 'dummy',
-      version: '1.0.0',
-    });
-
-    // Incorrect version
-    port1.postMessage({
-      arguments: [1, 2],
-      method: 'plusOne',
-      protocol: 'frame-rpc',
-      version: 'dummy',
-    });
-
-    // All incorrect
-    port1.postMessage({});
-    port1.postMessage(null);
-    port1.postMessage(undefined);
-    port1.postMessage(0);
-    port1.postMessage('');
-    port1.postMessage('dummy');
-
+    // All messages posted in the same turn of the event loop will get delivered
+    // by the time a setTimeout(..., 0) expires
     setTimeout(() => {
       assert.calledOnce(plusOne);
       done();
-    }, 100);
+    }, 0);
   });
+
+  [
+    {
+      message: {
+        arguments: 'test',
+        method: 'plusOne',
+        protocol: 'frame-rpc',
+        version: '1.0.0',
+      },
+      reason: 'message has incorrect arguments',
+    },
+
+    {
+      message: {
+        arguments: [1, 2],
+        method: 'dummy',
+        protocol: 'frame-rpc',
+        version: '1.0.0',
+      },
+      reason: 'message has incorrect method',
+    },
+    {
+      message: {
+        arguments: [1, 2],
+        method: 'plusOne',
+        protocol: 'dummy',
+        version: '1.0.0',
+      },
+      reason: 'message has incorrect protocol',
+    },
+
+    {
+      message: {
+        arguments: [1, 2],
+        method: 'plusOne',
+        protocol: 'frame-rpc',
+        version: 'dummy',
+      },
+      reason: 'message has incorrect version',
+    },
+    { message: {}, reason: 'message is an empty object' },
+    { message: null, reason: 'message is `null`' },
+    { message: undefined, reason: 'message is `undefined`' },
+    { message: 0, reason: 'message is `0`' },
+    { message: '', reason: 'message is empty string' },
+    { message: 'dummy', reason: 'message is a string' },
```

Comment:
The `reason` argument here is not used. One way that we make it easy to identify which case failed for a parametrized test is to inject some disambiguation string (eg. test case index or `reason` string) into the test description (eg. ``it(`should not call method on invalid message (${reason})`, ...``).

Reply:
🤦🏼 so simple... I was trying to display the `reason` inside the `it` function, and it was obviously more challenging.